### PR TITLE
[#PAB-366] add download post endpoint

### PR DESCRIPTION
**Pull request by cyrusdobbs**
https://trello.com/c/2nNVAwIx/366-add-file-download-server-side-functionality-to-front-end

### Change description
Built out the download POST endpoint to return a file of the requested format.

- [X] Unit tests and other appropriate tests added or updated
- [X] README and other documentation has been updated / added (if needed)
- [X] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
_If manual testing is needed, give suggested testing steps_

### CODE REVIEW

File: app/main/data.py [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
-        endpoint (str): an API get data address
+    This function constructs the request URL using the provided hostname, endpoint, and query parameters (if any).
+    It sends an HTTP GET request to the constructed URL and handles any bad responses.
 
-    Returns:
-        data (json): data response in json format
+    :param hostname: The hostname or base URL of the remote server.
+    :param endpoint: The endpoint or path of the remote resource to retrieve.
+    :param query_params: Optional dictionary of query parameters to include in the request URL.
+    :return: The Flask Response object containing the response from the remote server.
```
Comment (gidsg) [inline app/main/data.py:17]: I think we mean request response object rather than flask

File: app/main/routes.py [hunk 2]
```diff
@@ -29,8 +34,35 @@ def download():
...
+        )
+
+        if file_format == "json":
+            file_content = io.BytesIO(json.dumps(response.text).encode("utf-8"))
+            mimetype = "application/json"
+        elif file_format == "xlsx":
+            file_content = io.BytesIO(response.content)
+            mimetype = (
+                "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
+            )
```
Comment (gidsg) [inline app/main/routes.py:58]: Could we use a Match Statement here on `file_format` instead of `if/else`?

File: app/main/routes.py [hunk 3]
```diff
@@ -29,8 +34,35 @@ def download():
...
+
+        response = get_response(
+            Config.DATA_STORE_API_HOST,
+            "/download",
+            query_params={"file_format": file_format},
+        )
+
+        if file_format == "json":
+            file_content = io.BytesIO(json.dumps(response.text).encode("utf-8"))
+            mimetype = "application/json"
```
Comment (gidsg) [inline app/main/routes.py:53]: I think we might be able to infer the mimetype from the API response headers, rather than having to hardcode it here?

Comment (cyrusdobbs) [reply app/main/routes.py:53]: ah nice idea

**Review by gidsg — APPROVED** [review]

**Pull request merged**